qiskit/providers/ibmq/api/clients/account.py

```diff
@@ -169,46 +173,57 @@ def job_submit(
 
         if not submit_info:
             # Submit Qobj via HTTP.
-            submit_info = self._job_submit_post(backend_name, qobj_dict, job_name)
+            submit_info = self._job_submit_post(backend_name, qobj_dict, job_name, job_share_level)
 
         return submit_info
 
     def _job_submit_post(
             self,
             backend_name: str,
             qobj_dict: Dict[str, Any],
-            job_name: Optional[str] = None
+            job_name: Optional[str] = None,
+            job_share_level: Optional[ApiJobShareLevel] = None
     ) -> Dict[str, Any]:
         """Submit a Qobj to a device using HTTP POST.
 
         Args:
             backend_name: the name of the backend.
             qobj_dict: the Qobj to be executed, as a dictionary.
             job_name: custom name to be assigned to the job.
+            job_share_level: level the job should be shared at.
 
         Returns:
             job status.
         """
-        return self.client_api.job_submit(backend_name, qobj_dict, job_name)
+        return self.client_api.job_submit(
+            backend_name,
+            qobj_dict,
+            job_name,
+            job_share_level=getattr(job_share_level, 'value', None))
```

Comment:
I personally don't like using `getattr` with a hard coded attribute name because it's hard to catch if the variable type changes (probably not a risk here). But I don't feel too strongly about this one if you want to keep it.

Reply:
I was slightly skeptical when deciding to use `getattr` as well, but chose to go with it in the likes of keeping everything a bit more "tidy". But, I don't think it would be a big deal if we change it to access the attribute more directly.